alibi/datasets/default.py

```diff
@@ -67,7 +66,45 @@ def fetch_imagenet_10(url_id: int = 0) -> Dict:
     except RequestException:
         logger.exception("Could not connect, URL may be out of service")
         raise
-    imagenet10 = joblib.load(urllib.request.urlopen(url))
+    tar = tarfile.open(fileobj=BytesIO(resp.content), mode="r:gz")
+    imagenet10 = {}
```

Comment:
I would move this to bottom where the return dictionary is created for clarity. Or consider even building it in one go, i.e.
```python
imagenet10 = {
    'trainset': (X_train, y_train),
    ...
}
```